crates/sui-core/src/consensus_adapter.rs

```diff
@@ -569,16 +569,23 @@ impl ConsensusAdapter {
             Either::Right(((), processed_waiter)) => Some(processed_waiter),
         };
         let transaction_key = transaction.key();
-        let _monitor = CancelOnDrop(spawn_monitored_task!(async {
-            let mut i = 0u64;
-            loop {
-                i += 1;
-                const WARN_DELAY_S: u64 = 30;
-                tokio::time::sleep(Duration::from_secs(WARN_DELAY_S)).await;
-                let total_wait = i * WARN_DELAY_S;
-                warn!("Still waiting {total_wait} seconds for transaction {transaction_key:?} to commit in narwhal");
-            }
-        }));
+        // Log warnings for capability or end of publish transactions that fail to get sequenced
+        if matches!(
+            transaction.kind,
+            ConsensusTransactionKind::EndOfPublish(_)
+                | ConsensusTransactionKind::CapabilityNotification(_)
+        ) {
+            let _monitor = CancelOnDrop(spawn_monitored_task!(async {
```

Comment:
You need to move _monitor outside of this code block, otherwise it is dropped right away and has no effect

Reply:
Ouch, right, good catch